File: src/main/java/com/claudio_personagens/bookingcharactersapi/services/ClientServiceImpl.java
package com.claudio_personagens.bookingcharactersapi.services;

import com.claudio_personagens.bookingcharactersapi.domain.Client;
import com.claudio_personagens.bookingcharactersapi.repositories.ClientRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ClientServiceImpl  implements ClientService {

    @Autowired
    private ClientRepository clientRepository;

    @Override
    public Client save(Client client) {
        return clientRepository.save(client);
    }

    @Override
    public Optional<Client> findByName(String name) {
        return clientRepository.findByName(name);
    }

    @Override
    public Optional<Client> findById(String id) {
        return clientRepository.findById(id);
    }

    @Override
    public List<Client> findAll() {
        return clientRepository.findAll();
    }
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/services/ClientServiceImpl.kt
package com.claudio_personagens.bookingcharactersapi.services

import com.claudio_personagens.bookingcharactersapi.base.BaseEntity
import com.claudio_personagens.bookingcharactersapi.domain.Client
import com.claudio_personagens.bookingcharactersapi.repositories.ClientRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Service
class ClientServiceImpl (
    private val clientRepository: ClientRepository
) : ClientService {

    @Autowired
    private lateinit var bCryptPasswordEncoder: BCryptPasswordEncoder

    @Transactional
    override fun save(client: Client) : Client {
        //client.password = bCryptPasswordEncoder.encode(client.password)
        return clientRepository.save(client)
    }

    override fun findByName(name: String) = clientRepository.findByName(name)

    override fun findById(id: UUID) = clientRepository.findById(id)

    override fun findAll(): MutableList<Client> = clientRepository.findAll()

    override fun deleteById(id: UUID) = clientRepository.deleteById(id)
}